### PR TITLE
Viite 1675 port 1657 to master

### PR DESCRIPTION
**Pull request by pedrosag**

### CODE REVIEW

**Comment by pedrosag** [comment]
Please do merge the Viite-1676 to master before attempting this one, thank you very much chaps.

File: viite-UI/src/view/ProjectLinkLayer.js [hunk 1]
```diff
@@ -729,7 +730,8 @@
       if (map.getView().getZoom() > zoomlevels.minZoomForDirectionalMarkers) {
         var addMarkersToLayer = function(links, layer) {
           var directionMarkers = _.filter(links, function (projectLink) {
-            return projectLink.floating !== SelectionType.Floating.value && projectLink.anomaly !== Anomaly.NoAddressGiven.value && projectLink.anomaly !== Anomaly.GeometryChanged.value && (projectLink.sideCode === SideCode.AgainstDigitizing.value || projectLink.sideCode === SideCode.TowardsDigitizing.value);
+              var acceptedLinks = projectLink.id !== 0 || (projectLink.id === 0 && projectLink.anomaly === Anomaly.NoAddressGiven.value) || (projectLink.id === 0 && projectLink.roadLinkType === RoadLinkType.FloatingRoadLinkType.value);
```
Comment (Campanha) [inline viite-UI/src/view/ProjectLinkLayer.js]: _Can be changed to_ 
var acceptedLinks = projectLink.id !== 0 || (projectLink.id === 0 && (projectLink.anomaly === Anomaly.NoAddressGiven.value || projectLink.roadLinkType === RoadLinkType.FloatingRoadLinkType.value));

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala [hunk 2]
```diff
@@ -118,36 +144,35 @@ class DefaultSectionCalculatorStrategy extends RoadAddressSectionCalculatorStrat
...
+        } else {
+          val sampleLink = remainLinks.head.geometry.head
+          if (remainLinks.map(p => GeometryUtils.geometryEndpoints(p.geometry)).forall(geo => geo._1.y == sampleLink.y && geo._2.y == sampleLink.y)) {
+            // Very rare case
+            // Start and end point of the geometry Y Values are equal on the tracks check the X Value
+            Vector3d(1.0, 0.0, 0.0)
+          } else {
+            // Use the default vector
+            Vector3d(0.0, 1.0, 0.0)
+          }
```
Comment (leirasf) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/process/strategy/DefaultSectionCalculatorStrategy.scala]: The direction used on the `pl.getEndPoints(direction)` can not be the same if exists at least one project link with all points in the same Y.

**Comment by skosonen** [comment]
Tests pass on Jenkins. Check build #80.

**Pull request merged**